include/nbl/video/IGPUCommandPool.h

```diff
@@ -39,7 +39,6 @@ class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
         friend class CCommandSegment;
 
     public:
-        virtual ~ICommand() {}
```

Comment:
I believe this needs to be virtual, otherwise you'll have a problem with `~IFixedSizeCommand` (or any other derivative) not being called when you delete through an `ICommand` reference or pointer

Reply:
umm

```
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h: In destructor ‘virtual nbl::video::IGPUCommandPool::IFixedSizeCommand<nbl::video::IGPUCommandPool::CBindIndexBufferCmd>::~IFixedSizeCommand()’:
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h:74:25: error: use of deleted function ‘static void nbl::video::IGPUCommandPool::ICommand::operator delete(void*, std::align_val_t)’
```

Reply:
ok https://stackoverflow.com/questions/31686508/why-is-delete-operator-required-for-virtual-destructors

I think one of the "deleted" delete operators below needs to be uncommented (unsure which one), as you shouldn't be able to `new` or `delete` an ICommand, only placement new (no allocation or deallocation of memory)

Reply:
once this gets resolved I'll merge

Reply:
I looked through this a bit: https://en.cppreference.com/w/cpp/memory/new/operator_delete

I originally tried to make delete private but I guess they still get implemented even if your not using those delete methods?
```
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h: In instantiation of ‘class nbl::video::IGPUCommandPool::IFixedSizeCommand<nbl::video::IGPUCommandPool::CBeginCmd>’:
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h:360:60:   required from here
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h:78:25: error: deleted function ‘virtual nbl::video::IGPUCommandPool::IFixedSizeCommand<nbl::video::IGPUCommandPool::CBeginCmd>::~IFixedSizeCommand()’ overriding non-deleted function
[build]    78 |     class NBL_FORCE_EBO IFixedSizeCommand : public IGPUCommandPool::ICommand
[build]       |                         ^~~~~~~~~~~~~~~~~
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h:42:17: note: overridden function is ‘virtual constexpr nbl::video::IGPUCommandPool::ICommand::~ICommand()’
[build]    42 |         virtual ~ICommand() = default;
[build]       |                 ^
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h:78:25: note: ‘virtual nbl::video::IGPUCommandPool::IFixedSizeCommand<nbl::video::IGPUCommandPool::CBeginCmd>::~IFixedSizeCommand()’ is implicitly deleted because the default definition would be ill-formed:
[build]    78 |     class NBL_FORCE_EBO IFixedSizeCommand : public IGPUCommandPool::ICommand
[build]       |                         ^~~~~~~~~~~~~~~~~
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h:78:25: error: ‘static void nbl::video::IGPUCommandPool::ICommand::operator delete(void*)’ is private within this context
[build] /home/michaelpollind/projects/Nabla/include/nbl/video/IGPUCommandPool.h:62:21: note: declared private here
[build]    62 |         static void operator delete  (void* ptr) { assert(false); }
[build]       |                     ^~~~~~~~
```

would it be sufficient to make it protected and throw an assert if you try to use delete. I guess its not ideal since its failure at runtime.

Reply:
> I guess its not ideal since its failure at runtime.

no, figure out which one it uses during a `~ICommand`